### PR TITLE
add ability to specify single target using --target

### LINKED ISSUE #29: add --target option
currently, --target-file is the only way to specify targets to scan.  This is cumbersome when there's only a single IP/domain to scan.  

--target should take a single domain/ipv4/ipv6 address as an argument and scan that target without the user having to make a targetfile.  

the easiest way would be to process the argument and then write the file for the user in the place where the pipeline expects it to be.  

--target and --target-file should be mutually exclusive options

### PR DESCRIPTION
**Pull request by GreaterGoodest**
Adds ability to run scan on single domain/ip by creating a temporary target folder.
All tests passed.

This closes #29

### CODE REVIEW

File: pipeline/recon-pipeline.py [hunk 1]
```diff
@@ -235,6 +235,16 @@ def do_scan(self, args):
...
         command = ["luigi", "--module", scans.get(args.scantype)[0]]
 
+        if args.target and args.target_file:
+            self.async_alert(style(f"[!] You can't use the --target and --target-file options in the same scan!"))
+            return
+        elif args.target:
+            with open("tmp_tgt", "w") as file:
+                file.write(args.target)
+                args.__statement__.arg_list[args.__statement__.arg_list.index("--target") + 1] = os.path.abspath(
+                    "tmp_tgt"
```
Comment (epi052) [inline pipeline/recon-pipeline.py:245]: I like the idea of a temp file, but i'm not crazy about it being hard-coded.  I'm also not a fan of it being dropped in the user's cwd.  Take a look at [mkstemp](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp).  That will drop it in /tmp and give it a random name.  

Should probably be good stewards and do a check around [here](https://github.com/GreaterGoodest/recon-pipeline/blob/add-target/pipeline/recon-pipeline.py#L271) for `args.target` and if found clean up the file created by `mkstemp`.

File: pipeline/recon-pipeline.py [hunk 2]
```diff
@@ -235,6 +235,16 @@ def do_scan(self, args):
         # luigi --module pipeline.recon.web.webanalyze WebanalyzeScan --target-file tesla --top-ports 1000 --interface eth0
         command = ["luigi", "--module", scans.get(args.scantype)[0]]
 
+        if args.target and args.target_file:
+            self.async_alert(style(f"[!] You can't use the --target and --target-file options in the same scan!"))
+            return
+        elif args.target:
+            with open("tmp_tgt", "w") as file:
+                file.write(args.target)
+                args.__statement__.arg_list[args.__statement__.arg_list.index("--target") + 1] = os.path.abspath(
```
Comment (epi052) [inline pipeline/recon-pipeline.py:244]: This project uses pathlib for dealing with file system paths.  For consistency, can you please swap `os.path.abspath` to its pathlib equivalent?  Additionally, add a `.expanduser()` call before `.resolve()` as both `.resolve` and `.abspath` can have surprising results.  There's an example [here](https://github.com/GreaterGoodest/recon-pipeline/blob/add-target/pipeline/recon-pipeline.py#L429)

```
>>> os.path.abspath("~/stuff")
'/tmp/recon-pipeline/~/stuff'
>>> Path("~/stuff").resolve()
PosixPath('/tmp/recon-pipeline/~/stuff')
```

File: pipeline/recon-pipeline.py [hunk 3]
```diff
@@ -235,6 +235,16 @@ def do_scan(self, args):
         # luigi --module pipeline.recon.web.webanalyze WebanalyzeScan --target-file tesla --top-ports 1000 --interface eth0
         command = ["luigi", "--module", scans.get(args.scantype)[0]]
 
+        if args.target and args.target_file:
+            self.async_alert(style(f"[!] You can't use the --target and --target-file options in the same scan!"))
+            return
+        elif args.target:
+            with open("tmp_tgt", "w") as file:
```
Comment (epi052) [inline pipeline/recon-pipeline.py:242]: For trivially small writes like this, pathlib can be more succinct. 

`Path("my_tmp_file").write_text(args.target)`

It does the same things as the context manager.  You'll need a reference to your file for passing the path into the `arg_list`, so you can split that statement into two parts

```
tgt_file = Path("my_tmp_file")
tgt_file.write_text(args.target)
```

File: pipeline/recon-pipeline.py [hunk 4]
```diff
@@ -235,6 +235,16 @@ def do_scan(self, args):
         # luigi --module pipeline.recon.web.webanalyze WebanalyzeScan --target-file tesla --top-ports 1000 --interface eth0
         command = ["luigi", "--module", scans.get(args.scantype)[0]]
 
+        if args.target and args.target_file:
+            self.async_alert(style(f"[!] You can't use the --target and --target-file options in the same scan!"))
+            return
```
Comment (epi052) [inline pipeline/recon-pipeline.py:240]: mutual exclusion should be enforced through [argparse's add_mutually_exclusive_group](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group) when possible.

Also, either target-file or target is required, so `add_mutually_exclusive_group(required=True)` would be appropriate.  

Check [line 60 in recon.parsers](https://github.com/epi052/recon-pipeline/blob/store-results-in-a-database-project/pipeline/recon/parsers.py#L60) for an example.

**Comment by GreaterGoodest** [comment]
made requested changes, working on adding tests/documentation

**Comment by GreaterGoodest** [comment]
all requested changes should be g2g

**Comment by GreaterGoodest** [comment]
Requests integrated!

**Pull request merged**